### PR TITLE
Add table and sequence ownership to pgbedrock

### PR DESCRIPTION
**Pull request by zcmarine**
This PR addresses Issue #4 (Support table and sequence ownership).

It does so by doing 3 things:

1. Have pgbedrock generate add table and sequence ownership information into the spec that pgbedrock generates.
2. Have the spec_inspector module validate a variety of conditions that are necessary in order for us to manage ownerships: that a spec has no objects with multiple owners, has no objects with no owners, has no auto-dependent objects listed with owners, and has no objects in the spec that are not in the database. This allows pgbedrock configure to run without having to check a ton of edge cases while it executes.
3. Have pgbedrock configure manage table and sequence ownership.

I have added tests to validate that the above 3 items work as expected and have also verified that doing `pgbedrock generate` followed by `pgbedrock configure` both runs on our staging database and that it shows no changes.

I know that this PR is kind of large and painful: these 3 things all depend upon each other, so they need to all be merged into master concurrently. Ideally this might have been a feature branch with 3 separate PRs added to it. That didn't happen though, so if there's any way I can make this less painful to read through let me know.

### CODE REVIEW

File: pgbedrock/attributes.py [hunk 1]
```diff
@@ -60,7 +57,6 @@
 def analyze_attributes(spec, cursor, verbose):
     logger.debug('Starting analyze_attributes()')
     dbcontext = DatabaseContext(cursor, verbose)
-    fail_if_undocumented_roles(spec, dbcontext)
```
Comment (punkrockpolly) [inline pgbedrock/attributes.py]: what's the reason for removing this failure case?

File: tests/test_ownerships.py [hunk 2]
```diff
@@ -48,44 +32,122 @@ def test_analyze_schemas_create_schemas(cursor):
...
+    attributes.Q_CREATE_ROLE.format(ROLES[0]),
+    attributes.Q_CREATE_ROLE.format(ROLES[1]),
+    own.Q_CREATE_SCHEMA.format(SCHEMAS[0], ROLES[0]),
+    own.Q_CREATE_SCHEMA.format(SCHEMAS[1], ROLES[0]),
+    privileges.Q_GRANT_NONDEFAULT.format('CREATE', 'SCHEMA', SCHEMAS[0], ROLES[1]),
+    privileges.Q_GRANT_NONDEFAULT.format('CREATE', 'SCHEMA', SCHEMAS[1], ROLES[1]),
+
+    # Create tables in SCHEMAS[0], some of which aren't owned by ROLES[0]
+    Q_CREATE_TABLE.format(ROLES[0], SCHEMAS[0], TABLES[0]),
+    Q_CREATE_TABLE.format(ROLES[1], SCHEMAS[0], TABLES[1]),  #
```
Comment (punkrockpolly) [inline tests/test_ownerships.py]: got some dangling `#`

File: tests/test_ownerships.py [hunk 3]
```diff
@@ -48,44 +32,122 @@ def test_analyze_schemas_create_schemas(cursor):
...
+    Q_CREATE_TABLE.format(ROLES[1], SCHEMAS[0], TABLES[2]),  #
```
Comment (punkrockpolly) [inline tests/test_ownerships.py]: `#`

File: tests/test_spec_inspector.py [hunk 4]
```diff
@@ -45,12 +47,271 @@ def test_verify_spec_fails_multiple_roles_own_schema_personal_schema(capsys):
...
+            tables:
+                - schema0.table0
+                - schema1.table1
+
+    role1:
+        owns:
+            tables:
+                # Ensure function can handle some objects being quoted and some not
+                - schema1.table1
+                - schema1."table2"
```
Comment (punkrockpolly) [inline tests/test_spec_inspector.py]: question: are there certain names that **have** to be quoted in order to parse properly?

File: pgbedrock/attributes.py (hunk 1 above)
Comment (zcmarine) [reply pgbedrock/attributes.py]: It's moved into the spec_inspector.py, for a few reasons:
1. That way all spec validation code is in one module together
2. If there's an issue with the spec it'd be nice to fail immediately vs. go through a bunch of submodules and _then_ fail, which is what happens currently. By co-locating this all in the spec_inspector.py, when we load the spec we check the quality of the spec and can give that immediate feedback.

File: tests/test_spec_inspector.py (hunk 4 above)
Comment (zcmarine) [reply tests/test_spec_inspector.py]: Yeah, if you have a '.' in the table name. For example, there have historically been a few tables that would be like jdoe.jdoe.revenue_metrics. I believe Postgres is actually relatively permissive with table names so long as you throw the name within double quotes.

Comment (cpdean) [reply tests/test_spec_inspector.py]: Yep -- this is correct.  You're allowed to do lots of weird things in table/column identifiers as long as you wrap it in double quotes.

```
-- to make everyone uncomfortable when trying to use this in reports
SELECT
*
FROM
cpdean."please never use this it has lots of issues but approximates an important metric";
```
